Домашнее задание

NekhoroshkinaElena opened the pull request.

NekhoroshkinaElena committed:
Init commit

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,102 @@
+            android:textAllCaps="false"
+            android:textColor="@color/black"
+            android:textSize="@dimen/text_size_title"
+            app:cornerRadius="16dp"
```
Повторяющийся аналогичный размер, в ресурсы

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,102 @@
+            app:cornerRadius="16dp"
+            app:icon="@drawable/ic_settings_light_mode"
+            app:iconGravity="textStart"
+            app:iconPadding="8dp"
```
Повторяющийся аналогичный размер, в ресурсы

faritowich commented on app/src/main/java/com/example/playlistmaker/SettingsActivity.kt:
```diff
@@ -0,0 +1,19 @@
+        val buttonBack = findViewById<Button>(R.id.buttonBack)
+        buttonBack.setOnClickListener {
+            val displayIntent = Intent(this, MainActivity::class.java)
+            startActivity(displayIntent)
```
Для перехода назад надо использовать не интент, а методы finish() или onBackPressed(), потому что иначе вместо возврата в предыдущий экран создается новый экземпляр MainActivity и при возврате обратно на главный экран и нажатии на кнопку телефона "назад" мы "вернемся" снова на этот экран. На эту тему можно почитать про стек активити. Это будет критично в следующем спринте.

NekhoroshkinaElena replied:
Добрый день, сама заметила эту проблему, но побоялась использовать незнакомые инструменты, не зная точно, как они работают, спасибо за подсказку!

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,102 @@
+            android:textColor="@color/white"
+            android:textSize="@dimen/text_size_title" />
+
+    </FrameLayout>
```
Лишняя вложенность, FrameLayout не нужен

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,102 @@
+        android:layout_height="match_parent"
+        android:orientation="vertical"
+        android:paddingStart="16dp"
+        android:paddingEnd="16dp"
```
paddingStart + paddingEnd = paddingHorizontal

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,122 @@
+        android:layout_height="wrap_content"
+        android:orientation="vertical"
+        android:layout_marginTop="24dp">
+    <Switch
```
Поплыло форматирование. Есть хороший хоткей, можно нажать Alt + Ctrl + L и студия красиво отформатирует весь код на экране

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,122 @@
+        android:textAllCaps="false"
+        android:textAppearance="@style/TextAppearance.AppCompat.Small"
+        android:textColor="@color/black"
+        android:textSize="16sp"/>
```
@dimen/text_size_regular

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,122 @@
+        android:paddingStart="16dp"
+        android:paddingEnd="12dp"
+        android:paddingTop="21dp"
+        android:paddingBottom="21dp"
```
Рекомендуется делать размеры View и отступы кратными 8 (или 4). Это обеспечивает единообразие дизайна, а также связано с тем, что большинство размеров экрана делятся на 8. Иногда эту рекомендацию приходится нарушать, но лучше по возможности ей следовать.
Также см. https://material.io/design/layout/spacing-methods.html#baseline-grid

NekhoroshkinaElena committed:
refactor: вынести в ресурсы размеры, которые используются несколько раз
fix: использовать finish() вместо интент, для перехода на предыдущий экран
fix: избавиться от лишней вложенности FrameLayout в классе SettingsActivity
style: исправить форматирование в activity_settings.xml
refactor: сделать отступы кратные 4 или 8

faritowich commented on app/src/main/res/values/dimens.xml:
```diff
@@ -2,4 +2,11 @@
     <dimen name="text_size_regular">16sp</dimen>
+    <dimen name="icon_padding">8dp</dimen>
+    <dimen name="button_radius">16dp</dimen>
+    <dimen name="zero_value">0dp</dimen>
```
0dp обычно не выносят:)

NekhoroshkinaElena committed:
refactor: убрать нулевое значение из ресурсного файла
